### PR TITLE
feat(test): add test bundle

### PR DESCRIPTION
**Pull request by juliemr**
Create a bundle for the test library, TestComponentBuilder,
TestInjector, and DebugElement.

Internal tests use a superset of the bundle.

[Review on Reviewable](https://reviewable.io/reviews/angular/angular/2752)

### CODE REVIEW

**Comment by juliemr** [comment]
Pending #2749 to fix up the router tests.

**Comment by tbosch** [comment]
Note: @rkirov is out next week..

File: gulpfile.js [hunk 1]
```diff
@@ -909,7 +922,14 @@ gulp.task('bundle.js.sfx.dev.deps', ['bundle.js.sfx.dev'], function() {
...
 
-gulp.task('bundle.js.deps', ['bundle.js.prod.deps', 'bundle.js.dev.deps', 'bundle.js.min.deps', 'bundle.js.sfx.dev.deps', 'router.bundle.js.dev', 'mock.bundle.js.dev']);
+gulp.task('bundle.js.deps', [
+  'bundle.js.prod.deps',
+  'bundle.js.dev.deps',
+  'bundle.js.min.deps',
+  'bundle.js.sfx.dev.deps',
+  'router.bundle.js.dev',
+  'test.bundle.js.dev',
+  'mock.bundle.js.dev']);
```
Comment (rkirov) [inline gulpfile.js]: OOC, why can't we have mocks inside tests. Having too many bundles decreases the value of the bundles.

**Comment by btford** [comment]
+1 what Rado said: I think it's fine to have a single bundle.

I imagine that you'd often use mocks and tests together, and they'd never ship in production so you don't have to worry about the byte cost on download.

File: gulpfile.js (hunk 1 above)
Comment (juliemr) [inline gulpfile.js]: Done.

**Comment by juliemr** [comment]
Merged as 71c65b47f9c3cef4456355b8416df1211cee4836

**Pull request closed without merging**